src/pagination.middleware.ts

```diff
@@ -0,0 +1,27 @@
+import 'reflect-metadata';
```

Comment:
Do we need this here? I'm not finding a need...

Reply:
> I think there are some unneeded imports of reflect-metadata

I was getting a lot of error like

```
Uncaught TypeError: Reflect.getMetadata is not a function
```

So this helped me suppress it... found this solution [here](https://github.com/AngularClass/angular-starter/issues/1238#issuecomment-268723687) but do not know a better alternative.

Do you know the root cause for it? Any idea if something at `tsconfig` might fix this issue without these imports?

> `ArgumentArg`... a little bit strange but I have no better suggestion

Same here 😞

Reply:
@tibawatanabe 

> I think there are some unneeded imports of reflect-metadata

Ok, I did some reading and I need it just in my app/lib entry point, so I added it to both `index.ts` and `*.spec.ts`